lib/ansible/plugins/netconf/__init__.py

```diff
@@ -223,9 +223,14 @@ def dispatch(self, rpc_command=None, source=None, filter=None):
         """
         if rpc_command is None:
             raise ValueError('rpc_command value must be provided')
+        req = fromstring(rpc_command)
+        resp = self.m.dispatch(req, source=source, filter=filter)
 
-        resp = self.m.dispatch(fromstring(rpc_command), source=source, filter=filter)
-        return resp.data_xml if hasattr(resp, 'data_xml') else resp.xml
+        result = resp.xml
```

Comment:
better to move this line in else part. something like
```
        if hasattr(resp, 'data_xml') or hasattr(resp, 'data_ele'):
            result = resp.data_xml
       else
            result = resp.xml
```

Reply:
sure, why better? Should be the same output right?

Reply:
In most cases data_xml object is present the resp object, returning `xml` object is catch all type of scenario, hence in former code result variable is assigned twice most of the times.